x-pack/plugins/alerting/server/task_runner/create_execution_handler.ts

```diff
@@ -193,6 +198,28 @@ export function createExecutionHandler<
           : `actionGroup: '${actionGroup}'`
       } action: ${actionLabel}`;
       eventLogger.logEvent(event);
+      // resolve(true);
+      // } catch (err) {
+      //   return reject(err);
+      // }
+      // });
+      // promises.push(promise);
+      // await actionsClient.executeEphemeralTask({
+      //   taskType: `actions:${action.actionTypeId}`,
+      //   params: {
+      //     ...action.params,
+      //     taskParams: {
+      //       actionId: action.id,
+      //       apiKey,
+      //     },
+      //   },
+      //   state: {},
+      // });
     }
+
+    const actionsClient = await actionsPlugin.getActionsClientWithRequest(request);
+    await actionsClient.executeEphemeralTasks(tasks);
```

Comment:
We probably don't want to await here - using TM ensures we aren't going to actually release lots of work that can't be tracked, and so we can execute them and complete the Rule execution right away.

This ensures we reschedule the rule no matter how long it takes to run the actions.

Reply:
This makes sense. 

Should we wait to write to the action log until after the action completes? This will make testing easier, as the tests are setup to watch for event logs

Reply:
Nvm. I got clarify on the difference between `alerting:execute-action` and `actions:execute` in terms of the event log and I updated this a bit.